src/cplscheme/BaseCouplingScheme.cpp

```diff
@@ -280,9 +280,15 @@ bool BaseCouplingScheme::hasTimeWindowSize() const
   return not math::equals(_timeWindowSize, UNDEFINED_TIME_WINDOW_SIZE);
 }
 
+bool BaseCouplingScheme::solverSetsTimeWindowSize() const
+{
+  PRECICE_ASSERT(hasTimeWindowSize());
+  return false;
+}
```

Comment:
Isn't this function a copy of `hastTimeWindowSize`?
I agree that the name `solverSetsTimeWindowSize()` is much clearer.

Reply:
> Isn't this function a copy of `hastTimeWindowSize`?

From the perspective of functionality: Yes. But I think it clearly improves readability to have two functions here. It's from the outside perspective not obvious that `hasTimeWindowSize() == !solverSetsTimeWindowSize()`.

If we remove `hasTimeWindowSize`, we would end up in some situations with something like this:

```
double BaseCouplingScheme::getThisTimeWindowRemainder() const
{
  PRECICE_TRACE();
  double remainder = 0.0;
  if (!solverSetsTimeWindowSize()) {
    remainder = getNextTimestepMaxLength();
  }
  PRECICE_DEBUG("return {}", remainder);
  return remainder;
}
```

instead of 

```
double BaseCouplingScheme::getThisTimeWindowRemainder() const
{
  PRECICE_TRACE();
  double remainder = 0.0;
  if (hasTimeWindowSize()) {
    remainder = getNextTimestepMaxLength();
  }
  PRECICE_DEBUG("return {}", remainder);
  return remainder;
}
```

So I would like to keep both functions for the sake of readability.

Reply:
Mmh, I actually find the first version easier to read. It gives additional information on when what happens.
In then end, `hasTimeWindowSize` is hard to understand, what does "has a time window size" mean?